File: 1299-replace-elements-with-greatest-element-on-right-side/1299-replace-elements-with-greatest-element-on-right-side.cpp
class Solution {
public:
    vector<int> replaceElements(vector<int>& arr) {
        int n = arr.size();
        int maximum = arr[n-1];
        arr[n-1] = -1;
        for (int i=n-2; i>=0;i--){
            int temp = arr[i];
            arr[i] = maximum;
            maximum = max(arr[i], temp);
        }
        return arr;
    }
};

File: 1299-replace-elements-with-greatest-element-on-right-side/1299-replace-elements-with-greatest-element-on-right-side.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        n = len(arr)
        maximum = arr[n-1]
        arr[n-1] = -1
        for i in range(n-2, -1, -1):
            temp = arr[i]
            arr[i] = maximum
            maximum = max(maximum, temp)
        return arr
                    
                
        